### PR TITLE
config: add server level isolation read config

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
For isolation read. We have the global and session variable `tidb_isolation_read_engines` to control. But we can't control specific tidb-server to set isolation read. So we need a server-level config to control the isolation.

### What is changed and how it works?
When both the server-level isolation read and session-level isolation read exists. The access paths should be filtered by both of them.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Related changes

 - Need to update the documentation

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -367,3 +367,10 @@ history-size = 24
 [experimental]
 # enable column attribute `auto_random` to be defined on the primary key column.
 allow-auto-random = false
+
+# server level isolation read by engines and labels
+[isolation-read]
+# engines means allow the tidb server read data from which types of engines. options: "tikv", "tiflash", "tidb".
+engines = ["tikv", "tiflash", "tidb"]
+# labels means allow the tidb server read data from which labels. Empty set means all labels are allowed.
```
Comment (eurekaka) [inline config/config.toml.example]: Could you explain more on the purpose of `labels` in the comment? or the use cases of the `labels`?

File: config/config.go [hunk 2]
```diff
@@ -585,6 +593,9 @@ var defaultConf = Config{
 		RefreshInterval: 1800,
 		HistorySize:     24,
 	},
+	IsolationRead: IsolationRead{
+		Engines: []string{"tikv", "tiflash", "tidb"},
```
Comment (lamxTyler) [inline config/config.go]: Add some config check?

**Comment by lzmhhh123** [comment]
@eurekaka @lamxTyler @winoros PTAL.

File: planner/core/planbuilder.go [hunk 3]
```diff
@@ -730,6 +730,10 @@ func (b *PlanBuilder) filterPathByIsolationRead(paths []*util.AccessPath, dbName
 	if dbName.L == mysql.SystemDB {
 		return paths, nil
 	}
+	cfgIsolationEngines := make(map[string]struct{})
+	for _, engine := range config.GetGlobalConfig().IsolationRead.Engines {
+		cfgIsolationEngines[engine] = struct{}{}
```
Comment (winoros) [inline planner/core/planbuilder.go]: This map can be initialized when we set the config so we don't need to always create a new map when we complie one SQL?

Comment (lzmhhh123) [reply planner/core/planbuilder.go]: For the server config. It's hard to create a map by `toml`.

Comment (lzmhhh123) [reply planner/core/planbuilder.go]: Do you have any advice?

Comment (eurekaka) [reply planner/core/planbuilder.go]: Make it a global variable and init it in places around `Config::Valid()` for example?

Comment (lzmhhh123) [reply planner/core/planbuilder.go]: I prefer not to do this. There are two reasons:
1. If we set the map in `Config::Valid()`, there are many places of tests need to keep consistent for `engines` and `EnginesMap`.
2. The size of the `Engines` array is very little. At most three. I think the optimization is unnecessary.

Comment (lonng) [reply planner/core/planbuilder.go]: Use `set.StringSet` is better.

**Pull request merged**